API_CONVENTIONS.md

```diff
@@ -0,0 +1,43 @@
+# API design conventions
```

Comment:
nit: is there a reason for the all caps name (API_CONVENTIONS.md)? If not, I'd switch it to lowercase

Reply:
The main reason is to match the casing of markdown files in the root directory.